gph/python/ripser_interface.py

```diff
@@ -106,6 +107,21 @@ def _collapse_coo(row, col, data, thresh):
             np.hstack([data_diag, data]))
 
 
+def _collapse_dense(dm, thresh):
+    """Run edge collapser on off-diagonal data and then reinsert diagonal
+    data if any non zero values is present"""
+    row, col, data = gph_collapser.flag_complex_collapse_edges_dense(
+        dm.astype(np.float32), thresh)
+
+    if (dm.diagonal() != 0).any():
+        indices = np.arange(dm.diagonal().shape[0])
+        row = np.hstack([indices, row])
+        col = np.hstack([indices, col])
+        data = np.hstack([dm.diagonal(), data])
```

Comment:
```suggestion
    data_diag = dm.diagonal() != 0
    if (data_diag != 0).any():
        indices = np.arange(data_diag.shape[0])
        row = np.hstack([indices, row])
        col = np.hstack([indices, col])
        data = np.hstack([data_diag, data])
```
to compute `dm.diagonal()` once instead of three times (and, to me at least, make the code slightly more readable).